packages/services/api/src/modules/target/resolvers/Mutation/updateTargetDangerousChangeClassification.ts

```diff
@@ -0,0 +1,44 @@
+import { IdTranslator } from '../../../shared/providers/id-translator';
+import { TargetManager } from '../../providers/target-manager';
+import { MutationResolvers } from './../../../../__generated__/types';
+
+export const updateTargetDangerousChangeClassification: NonNullable<
+  MutationResolvers['updateTargetDangerousChangeClassification']
+> = async (_, { input: { failDangerousChecks, target: targetRef } }, { injector }) => {
+  const targetManager = injector.get(TargetManager);
+  let targetId: string, projectId: string, organizationId: string;
+  if (targetRef.byId) {
+    const target = await targetManager.getTargetById({ targetId: targetRef.byId });
+    targetId = target.id;
+    projectId = target.projectId;
+    organizationId = target.orgId;
+  } else if (targetRef.bySelector) {
+    const translator = injector.get(IdTranslator);
+    [organizationId, projectId, targetId] = await Promise.all([
+      translator.translateOrganizationId(targetRef.bySelector),
+      translator.translateProjectId(targetRef.bySelector),
+      translator.translateTargetId(targetRef.bySelector),
+    ]);
+  } else {
+    throw new Error();
+  }
```

Comment:
Shouldn't this use the `resolveTargetReference` method from `packages/services/api/src/modules/shared/providers/id-translator.ts` ?

Reply:
Also I would move this to the `target-manager` `updateTargetDangerousChangeClassification` method 🤔

Reply:
`resolveTargetReference` is very handy. I had forgot it existed. Ty.